src/cow-react/modules/swap/pure/EthFlow/EthFlowBanner/index.tsx

```diff
@@ -0,0 +1,145 @@
+  > strong {
+    font-weight: 600;
+  }
+  > img:first-child {
+    filter: ${({ theme }) => `invert(${theme.darkMode ? 0 : 1})`};
+  }
+  > svg:last-child {
+    cursor: pointer;
+  }
+  > svg,
+  > strong {
+    margin: auto;
+  }
+`
+
+const BannerInnerWrapper = styled.div`
+  display: grid;
+  grid-template-rows: auto;
+  align-items: center;
+  justify-content: stretch;
+  width: 100%;
+  text-align: left;
+  > p {
+    padding: 0 10px;
+    margin-bottom: 0;
+  }
+`
+
+const WrapEthCta = styled(BannerInnerWrapper)`
+  flex-flow: row nowrap;
+  text-align: center;
+  > span {
+    cursor: pointer;
+    font-weight: bold;
+    margin: 12px 0 4px;
+  }
+`
+
+const SpanCta = styled.span`
+  > small {
+    display: block;
+    margin: 5px 0 0;
+    font-weight: 400;
+  }
+`
+
+export interface EthFlowBannerContentProps extends EthFlowBannerCallbacks {
+  native: Currency
+  wrapped: Token & { logoURI: string }
+  showBanner: boolean
+  hasEnoughWrappedBalance: boolean
+  showBannerCallback: () => void
+}
+
+export function EthFlowBannerContent(props: EthFlowBannerContentProps) {
+  const {
+    native,
+    wrapped,
+    showBanner,
+    hasEnoughWrappedBalance,
+    showBannerCallback,
+    switchCurrencyCallback,
+    wrapCallback,
+    forceWrapCallback,
+  } = props
+  return (
+    <BannerWrapper onClick={showBannerCallback}>
+      <ClosedBannerWrapper>
+        <img alt="eth-flow-icon" src={ethFlowIcon} />
+        <strong>
+          {hasEnoughWrappedBalance
+            ? `Switch to the classic ${wrapped.symbol} experience and benefit!`
+            : `Wrap your ${native.symbol} and use the classic ${wrapped.symbol} experience!`}
+        </strong>
+        {showBanner ? <ChevronUp size={20} /> : <ChevronDown size={20} />}
+      </ClosedBannerWrapper>
+      {showBanner && (
+        <BannerInnerWrapper>
+          <p>
+            {!hasEnoughWrappedBalance && (
+              <>
+                You will be prompted to{' '}
+                <strong>
+                  wrap your {native.symbol} to {wrapped.symbol}
+                </strong>{' '}
+                before placing your order.
+              </>
+            )}{' '}
+            This way, you&apos;ll take of advantage of:
+          </p>
+          <ul>
+            <li>Lower overall fees</li>
+            <li>Lower default slippage (instead of {ETH_FLOW_SLIPPAGE.toSignificant(PERCENTAGE_PRECISION)}%)</li>
+            <li>No fees for failed transactions</li>
+          </ul>
+          <Separator />
+          <WrapEthCta>
+            {hasEnoughWrappedBalance ? (
+              <>
+                <SpanCta onClick={switchCurrencyCallback}>
+                  Switch to {wrapped.symbol}{' '}
+                  <small>
+                    or{' '}
+                    <u>
+                      {' '}
+                      <SpanCta onClick={forceWrapCallback}>wrap my {native.symbol} and swap</SpanCta>
+                    </u>
+                  </small>
```

Comment:
I think its really cool, that you thought even about this case, where a user still wants to wrap. 

But I would say that this is a very special case that likely won't happen frequently. I would be fine with leaving it out.
Then we would:
- reduce complexity,
- Make the "switch to {wrapped.symbol}{' '}" easier clickable.

@michelbio what do you think?

Reply:
all good, removing. as discussed, it is easier and the force wrap is unnecessary